Forgot Password

melcidar opened the pull request:
--------------------------------------------------------------------
 MAIN
 * this is the look of forgot password page --------------------------------------------------------------------
 OTHER
 * ignore changes in actions, because nothing is set --------------------------------------------------------------------
 
![image](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/119449806/6e478600-9b50-4898-9b89-e8b8d6c7b119)

* the look of reset password
![image](https://github.com/becarselman/ant-colony-pharaoh-frontend/assets/119449806/72a2cb06-78f3-4534-8495-52f238466a36)

melcidar committed:
Forgot Password - wip

--------------------------------------------------------------------
 MAIN
 * this is the look of forgot password page
--------------------------------------------------------------------
 OTHER
 * ignore changes in actions, because nothing is set
--------------------------------------------------------------------

melcidar committed:
Forget Password

--------------------------------------------------------------------
 MAIN
 * added sendPasswordResetEmail
--------------------------------------------------------------------
 OTHER
 * ///////
--------------------------------------------------------------------

melcidar committed:
Forgot Password

--------------------------------------------------------------------
 MAIN
 * tryed to implement function again due to errors
--------------------------------------------------------------------
 OTHER
 * /////
--------------------------------------------------------------------

melcidar committed:
FORGOT PASSWORD

--------------------------------------------------------------------
 MAIN
 * added reducer, changed function
--------------------------------------------------------------------
 OTHER
 * still doesn't want to send API request when button is clicked
--------------------------------------------------------------------

becarselman commented on src/components/loginform/LoginForm.js:
```diff
@@ -74,7 +74,7 @@ function LoginForm({ actions }) {
            <label className="remember-password" htmlFor="remember-password">Remember password</label>
            </div>
-            <a className="forgot-password" href="#">Forgot password?</a>
+           <a className="forgot-password" onClick={() => navigate('/forgot-password')} href="#">Forgot password?</a>
```
why you using a tag and then onClick. You can specify route in href and it will navigate to /forgot-password

becarselman commented on src/sagas/forgotPasswordSaga.js:
```diff
@@ -0,0 +1,19 @@
+
+function* forgotPasswordRequest({ payload }) {
+  const { email } = payload;
+  console.log("aaaaaaaaa");
```
remove console log

becarselman commented on src/sagas/forgotPasswordSaga.js:
```diff
@@ -0,0 +1,19 @@
+    yield call(sendForgotPasswordRequest, email);
+    yield put(forgotPasswordSuccess());
+  } catch (error) {
+    console.error('An error occurred:', error);
```
remove conosle.error

becarselman replied:
If you want to display error you can use notification popups.

melcidar committed:
FORGOT PASSWORD

--------------------------------------------------------------------
 MAIN
 * fixed code based on the comments on last commit
--------------------------------------------------------------------
 OTHER
 * still issues with sending API request after clicking on button
 * to be fixed soon, hopefully
--------------------------------------------------------------------

melcidar committed:
FORGOT PASSWORD

--------------------------------------------------------------------
 MAIN
 * fixed some issues, it should work now
--------------------------------------------------------------------
 OTHER
 * when "send me email" button is clicked, it gives error 404
 * which is fine, because we haven't connected frontend and backend
--------------------------------------------------------------------

melcidar committed:
RESET PASSWORD

--------------------------------------------------------------------
 MAIN
 * added reset password functionality
--------------------------------------------------------------------
 OTHER
 * I am still fighting with pop out messages and Toastify,
 I will try to add them later on
 * code is tested, works!
--------------------------------------------------------------------

becarselman commented on src/components/forgotpassword/ForgotPassword.js:
```diff
@@ -0,0 +1,85 @@
+import { passwordRequest } from '../../actions/authActions.js';
+
+
+function ForgotPassword({ actions }) {
```
please use arrow function

becarselman commented on src/components/forgotpassword/ForgotPassword.js:
```diff
@@ -0,0 +1,85 @@
+    e.preventDefault();
+    try {
+      setLoading(true);
+      const response = await dispatch(passwordRequest({ email: email }));
```
we don't need dispatch because we have mapDispatchToProps in container.js

becarselman commented on src/components/resetpassword/resetPassword.js:
```diff
@@ -0,0 +1,104 @@
+    setErrorMessage('');
+  
+    try {
+      await dispatch(resetPasswordRequest(token, newPassword));
```
same as prev comment

becarselman commented on src/service/authService.js:
```diff
@@ -12,6 +13,43 @@ const login = (data) => {
+
+export async function resetPasswordRequest(token, newPassword) {
+  try {
+    const response = await axios.post(`http://localhost:5000/reset-password/${token}`, { newPassword }, {
```
use this url from env

melcidar committed:
Trying without dispatch

melcidar committed:
FORGOT PASSWORD AND RESET PASSWORD

--------------------------------------------------------------------
 MAIN
 * removed dispatch and used mapDispatchToProps part of code instead
 * for both forgot and reset password
--------------------------------------------------------------------
 OTHER
 * tested it with backend, it works.
--------------------------------------------------------------------

becarselman commented on src/components/forgotpassword/ForgotPassword.js:
```diff
@@ -0,0 +1,83 @@
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    console.log(actions)
```
remove console.log

melcidar committed:
FORGOT AND RESET PASSWORD

--------------------------------------------------------------------
 MAIN
 * removing console log
--------------------------------------------------------------------
 OTHER
 * ////
--------------------------------------------------------------------

melcidar committed:
Merge branch 'develop' into FEATURE/Meliha01/ForgotPasswordFrontend
